car_test.go

```diff
@@ -69,3 +73,79 @@ func TestRoundtrip(t *testing.T) {
 		}
 	}
 }
+
+func TestRoundtripSelective(t *testing.T) {
+	sourceBserv := dstest.Bserv()
+	sourceBs := sourceBserv.Blockstore()
+	dserv := dag.NewDAGService(sourceBserv)
+	a := dag.NewRawNode([]byte("aaaa"))
+	b := dag.NewRawNode([]byte("bbbb"))
+	c := dag.NewRawNode([]byte("cccc"))
+
+	nd1 := &dag.ProtoNode{}
+	nd1.AddNodeLink("cat", a)
+
+	nd2 := &dag.ProtoNode{}
+	nd2.AddNodeLink("first", nd1)
+	nd2.AddNodeLink("dog", b)
+	nd2.AddNodeLink("repeat", nd1)
+
+	nd3 := &dag.ProtoNode{}
+	nd3.AddNodeLink("second", nd2)
+	nd3.AddNodeLink("bear", c)
+
+	assertAddNodes(t, dserv, a, b, c, nd1, nd2, nd3)
+
+	ssb := builder.NewSelectorSpecBuilder(ipldfree.NodeBuilder())
+	selector := ssb.ExploreFields(func(efsb builder.ExploreFieldsSpecBuilder) {
+		efsb.Insert("Links",
+			ssb.ExploreIndex(1, ssb.ExploreRecursive(selector.RecursionLimitNone(), ssb.ExploreAll(ssb.ExploreRecursiveEdge()))))
+	}).Node()
+
+	sc := NewSelectiveCar(context.Background(), sourceBs, []Dag{Dag{Root: nd3.Cid(), Selector: selector}})
+
+	// write car in one step
+	buf := new(bytes.Buffer)
+	blockCount := 0
+	err := sc.Write(buf, func(block Block) error {
+		blockCount++
+		return nil
+	})
+	require.Equal(t, blockCount, 5)
+	require.NoError(t, err)
+
+	// write car in two steps
+	scp, err := sc.Prepare()
```

Comment:
should we reset this to a new `NewSelectiveCar()` since this is testing an entirely separate workflow?

Reply:
that's reasonable.